File: Project/pizza_restaurant.py
#create menu by dictionary
pizza_menu = {
    # Key: Category Name
    # Value: A sub-dictionary of Item: Price
    "crusts": {
        "thick": 120,   # Key: Item Name, Value: Price
        "thin": 100
    },

    "main toppings": {
        "seafood": 250,
        "hawaiian": 200,
        "meat lover": 280
    },

    "extra toppings": {
        "extra cheese": 50,
        "mushroom": 30,
        "bacon": 40
    }
}

# create function to order pizza
def pizza_restaurant():
    print("Welcome to the Pizza Restaurant!")
    user_name = input("Please enter your name: ")
    print(f"Hello, {user_name}! Let's order a pizza.")

    # use while loop to check invalid integer until enter valid integer
    while True: # while#1
        pizza_quantity = int(input("How many pizzas do you want to order?: "))
        if pizza_quantity > 0:
            break # if true run to 'for loop in pizza quantity'
        else: # if flase cont. while#1 loop
            print("Please enter a valid quantity.")
    
    # setting start of bill and list of order before start loop
    total_bill = 0             
    all_ordered_crusts = []    
    all_ordered_mains = []     
    all_final_toppings = [] 
    # then use for loop to order pizza as integer above
    for i in range(pizza_quantity):
        print(f"pizza# {i+1}")
        print("-"*40)
        
        # first: ask crust
        while True: # while#2
            crust_input = input("Please choose your crusts (thick/thin): ").lower()
            # use ifelse to check invalid value of crust
            if crust_input in pizza_menu["crusts"]:
                crust_price = pizza_menu["crusts"][crust_input]
                print(f"You have chosen {crust_input}: {crust_price} baht.")
                break # if true run to 'second'
            else: # if false cont while#2 loop
                print("Invalid crust choice.")

        # second: ask main toppings
        while True: # while#3
            maintop_input = input("Please choose your main toppings (seafood/hawaiian/meat lover): ").lower()
            # use ifelse to check invalid value of main topping
            if maintop_input in pizza_menu["main toppings"]:
                maintop_price = pizza_menu["main toppings"][maintop_input]
                print(f"You have chosen {maintop_input}: {maintop_price} baht.")
                break # if true run to 'third'
            else: # if false cont while#3 loop
                print("Invalid main topping choice.")

        # third: ask extra toppings
        # condition1: add or not
        # condition2: if added -> how many and loop in add toppings
        all_topping = []
        topping_total = 0
        while True: # while#4 to check condition y/n added
            topping_add = input("Do you want any toppings? Y/N: ")
            # condition1
            if topping_add.upper() == "Y": # if true run to condition2 while#5
                #use while loop to check invalid integer
                while True: # use while#5 loop to check int
                    # condition2
                    # how many
                    topping_quantity = int(input("How many toppings do you want to add?: "))
                    if topping_quantity > 0:
                        break # if true break this loop run 'for in topping quantity'
                    else: # if false cont while#5 loop
                        print("Please enter a valid quantity.")

                # loop in toppping quantity
                for j in range(topping_quantity):
                    while True: # use while#6 loop to check invalid topping input
                        topping_input = input("Please choose your extra toppings (extra cheese/mushroom/bacon): ").lower()
                        # use ifelse to check invalid value of extra topping
                        if topping_input in pizza_menu["extra toppings"]:
                            topping_price = pizza_menu["extra toppings"][topping_input]
                            topping_total += topping_price
                            all_topping.append(topping_input)
                            print(f"You have added {topping_input}: {topping_price} baht.")
                            break # if true break this loop run 'for in summary price'
                        else: # if false cont while#6 loop
                            print("Invalid extra topping choice.")
                break # break while#4 after finish for loop

            elif topping_add.upper() == "N":
                break # if true break condition1 run 'for in summary price'
            else: # if false not Y/N cont while#4
                print("Invalid choice. Please enter Y or N.")

        # accumulation price and add list after all loop above but in 1st for loop
        total_bill += (crust_price + maintop_price + topping_total) 
        all_ordered_crusts.append(crust_input)
        all_ordered_mains.append(maintop_input)
        all_final_toppings.extend(all_topping) # extend list to show
    
    # summary order
    print("\n" + "="*40)
    print("--- TOTAL ORDER SUMMARY ---")
    print(f"Total Pizzas: {pizza_quantity}")
    print(f"All Crusts: {', '.join(all_ordered_crusts)}")
    print(f"All Main Toppings: {', '.join(all_ordered_mains)}")
    print(f"All Extra Toppings: {', '.join(all_final_toppings) if all_final_toppings else 'None'}")
    print(f"GRAND TOTAL: {total_bill} Baht")
    print("="*40)

pizza_restaurant()

File: Project/pizza_restaurant.r
# menu items lists
dough <- c("classic", "thin")
flavor <- c("pepperoni", "bbq_chicken", "veggie_delight")
topping <- c("mushroom", "cheese", "chicken")
drinks <- c("cocacola", "water", "tea")
dessert <- c("tiramisu", "icecream", "lava_cake")
# price lists
dough_price <- c("classic" = 10, "thin" = 10)
flavor_price <- c("pepperoni" = 14, "bbq_chicken" = 15, "veggie_delight" = 13)
topping_price <- c("mushroom" = 1.5, "cheese" = 1.5, "chicken" = 1.5)
drinks_price <- c("cocacola" = 2.5, "water" = 2, "tea" = 3)
dessert_price <- c("tiramisu" = 5.5, "icecream" = 6, "lava_cake" = 4)

# start function pizza_restaurant
pizza_restaurant <- function() {
  bill <- 0 # set start bill = 0
  
  # get basic info & order by readline
  customer_name <- readline("Welcome to Pizza Paradise!, please give me your name: ")
  pizza_num <- as.numeric(readline("How many pizza do you want to order?(only number): "))
  # use is.na to recheck numeric if NA or < 1 can not continue next step
  if (is.na(pizza_num) | pizza_num < 1) {
        print("Warning: Invalid number of pizza. Order be cancelled.")
        pizza_num <- 0
      }
  pizza_counter <- 1
  
  # use while to run when all detail of pizza
  while (pizza_num > 0) {
    dough_order <- tolower(readline("Dough: Classic or Thin: "))
    flavor_order <- tolower(readline("Flavor (pepperoni, bbq_chicken, veggie_delight): "))
    print( paste("--Ordering Pizza#", pizza_counter,":" , dough_order, ",", flavor_order))
    pizza_cost <- 0
    
    # use if to check input names correctly with data in price
    if (dough_order %in% names(dough_price)) {
      pizza_cost <- pizza_cost + dough_price[dough_order]
    } else {
      print("Warning: Invalid dough. Using 'classic'.")
      pizza_cost <- pizza_cost + dough_price["classic"]
    }
    if (flavor_order %in% names(flavor_price)) {
      pizza_cost <- pizza_cost + flavor_price[flavor_order]
    } else {
      print( paste("Error!:", flavor_order, "Skipping this pizza."))
      pizza_num <- pizza_num -1 
      pizza_counter <- pizza_counter +1 
      next # use next to start next loop if condition not complete
    }
    
    # topping order using if for ask condition then using while loop to get input
    topping_add <- tolower(readline("Would you like to add any extra toppings? (y/n): "))
    if (topping_add == "y") {
      topping_num <- as.numeric(readline("How many topping do you want?(only number): "))
      if (is.na(topping_num) | topping_num < 1) {
        print("Warning: Invalid number of toppings. Setting topping to none.")
        topping_num <- 0
      }
      topping_counter <- 1
      while (topping_num > 0) {
        topping_order <- readline("Topping (mushroom, cheese, chicken): ")
        print( paste("--Topping added: ", topping_counter, topping_order))
        if (topping_order %in% names(topping_price)) {
          pizza_cost <- pizza_cost + topping_price[topping_order]
        } else {
          print( paste("Error!:", topping_order, "Skipping this topping."))
          topping_num <- topping_num -1
          topping_counter <- topping_counter +1
        }
        topping_num <- topping_num -1 # decrease every num to stop loop if num = input
        topping_counter <- topping_counter +1 # use to add number of each order
      }
    }
    bill <- bill + pizza_cost
    pizza_num <- pizza_num -1
    print( paste("--Pizza:Total cost:", round(bill,2))) # accumulate cost of all pizza
    pizza_counter <- pizza_counter +1
  }
  # add order drinks use the same syntax as above
  drinks_add <- tolower(readline("Anythings else? Do you want some drinks? (y/n): "))
  if (drinks_add == "y") {
    drinks_num <- as.numeric(readline("How many?(only number): "))
    if (is.na(drinks_num) | drinks_num < 1) {
      print("Warning: Invalid number of drinks. Setting drinks to none.")
      drinks_num <- 0
    }
    drinks_counter <- 1
    drinks_cost <- 0
    while (drinks_num > 0) {
      drinks_order <- tolower(readline("Drinks (cocacola, water, tea): "))
      if (drinks_order %in% names(drinks_price)) {
        drinks_cost <- drinks_cost + drinks_price[drinks_order]
      } else {
        print( paste("Error!:", drinks_order, "Skipping this drinks."))
        drinks_num <- drinks_num -1
        drinks_counter <- drinks_counter +1
      }
      drinks_num <- drinks_num -1
      drinks_counter <- drinks_counter +1
    } 
  }
  print( paste("Drinks cost:", round(drinks_cost,2)))
  drinks_counter <- drinks_counter +1
  # add order dessert
  dessert_add <- tolower(readline("Do you want some dessert? (y/n): "))
  if (dessert_add == "y") {
    dessert_num <- as.numeric(readline("How many?(only number): "))
    if (is.na(dessert_num) | dessert_num < 1) {
      print("Warning: Invalid number of dessert. Setting dessert to none.")
      dessert_num <- 0
    }
    dessert_counter <- 1
    dessert_cost <- 0
    while (dessert_num > 0) {
      dessert_order <- tolower(readline("Dessert (tiramisu, icecream, lava_cake): "))
      if (dessert_order %in% names(dessert_price)) {
        dessert_cost <- dessert_cost + dessert_price[dessert_order]
      } else {
        print( paste("Error!:", dessert_order, "Skipping this dessert."))
        dessert_num <- dessert_num -1
        dessert_counter <- dessert_counter +1
      }
      dessert_num <- dessert_num -1
      dessert_counter <- dessert_counter +1
    } 
  }
  print( paste("Dessert cost:", round(dessert_cost,2)))
  dessert_counter <- dessert_counter +1
  bill <- bill + drinks_cost + dessert_cost
  print("---Total bills---")
  print( paste0("$", bill))
  print("Thank you! We hope you enjoyed your meal.")

